internal/controllers/cluster/cluster_controller.go

```diff
@@ -62,19 +62,19 @@ const (
 // +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=clusters;clusters/status;clusters/finalizers,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=apiextensions.k8s.io,resources=customresourcedefinitions,verbs=get;list;watch
 
-// ClusterReconciler reconciles a Cluster object.
-type ClusterReconciler struct {
+// Reconciler reconciles a Cluster object.
+type Reconciler struct {
 	Client    client.Client
 	APIReader client.Reader
 
 	// WatchFilterValue is the label value used to filter events prior to reconciliation.
 	WatchFilterValue string
 
-	recorder        record.EventRecorder
+	Recorder        record.EventRecorder
 	externalTracker external.ObjectTracker
 }
 
-func (r *ClusterReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
+func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
```

Comment:
I guess this is technically a breaking change since ClusterReconciler does not exist for this package along with SetupWithManager() and Reconcile() so not sure if we want have any special consideration/deprecation.

Reply:
This not a breaking change, given that it happens in an internal package.
Also, in `controllers/alias.go` we still using `ClusterReconciler` as an alias for `internal/contolllers/cluster.Reconciler` (so e.g. no changes are required in main.go or in any other consumers for the existing types)

Reply:
I think is technically breaking in the sense that as consumer before this change I could have vendored the project, `import sigs.k8s.io/cluster-api/controllers` and call `ClusterReconciler{}.Reconcile()` / `MachineReconciler{}.Reconcile()` from my code. When vendoring this new changes the above will break my code.

I think this acceptable though and the potential exposure is pretty minimal.

Reply:
Agree, there is a breaking change as the Reconcile func does not exist anymore in the original package, but it's acceptable (and afaik covered by what we discussed in the CAPI meeting a while back)

Reply:
Eventually, we can expose also .Reconcile in alias.go, but If this is the case we should do in a separated PR applying the same change for all the other providers already moved to internal

Reply:
> Eventually, we can expose also .Reconcile in alias.go, but If this is the case we should do it in a separated PR applying the same change for all the other providers already moved to internal

Yup, but I would only want to do it if someone can explain why they are calling the reconcile func manually :)